sys/linux/usbip.txt

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <drivers/usb/usbip/usbip_common.h>
+
+resource usbip_server_fd[fd]
+
+syz_usbip_server_init(usb3 bool32, speed flags[usbip_device_speed]) usbip_server_fd
+write$usbip(fd usbip_server_fd, buffer ptr[in, usbip_header], len bytesize[buffer])
+
+type usbip_header_basic[PACKET_TYPE] {
+	command		const[PACKET_TYPE, int32be]
+	seqnum		int32be
+	devid		int32be
+	direction	flags[direction_flags, int32be]
+	ep		int32be
+} [packed]
+
+usbip_header_ret_submit {
+	status			int32be
+	actual_length		int32be
+	start_frame		int32be
+	number_of_packets	int32be[0:USBIP_MAX_ISO_PACKETS]
+	error_count		int32be
```

Comment:
Add padding fields to make the size of the header `== USBIP_HEADER_SIZE`.

But then also add a field for data at the end.

Reply:
https://elixir.bootlin.com/linux/latest/source/drivers/usb/usbip/vhci_rx.c#L55

Reply:
Unfortunately, any of these approaches doesn't work properly.  In this approach` make presubmit `fails with `sys/linux/usbip.txt:51:4: struct usbip_packet_ret_unlink has size attribute 1 which is less than struct size 24
` .

Reply:
Specifying size 1 for this struct is wrong. It is at least 24 bytes. Specify correct size and it should work.